.changeset/curly-bikes-train.md

```diff
@@ -0,0 +1,24 @@
+---
+"fluid-framework": minor
+"@fluidframework/tree": minor
+"__section": feature
+---
+Adds enablable allowed types to SchemaFactoryAlpha
+
+This adds the `enablable` API to [`SchemaFactoryAlpha`](https://fluidframework.com/docs/api/fluid-framework/schemafactoryalpha-class).
+Enablables can be used for schema evolution to add members to an [`AllowedTypes`](https://fluidframework.com/docs/api/fluid-framework/allowedtypes-typealias) while supporting cross version collaboration.
+
+Enablables are allowed types that can be read by the tree but not written. Attempting to write an enablable type will cause a runtime error.
```

Comment:
This need more clarity. If I have one client which has finished the migration (did a schema upgrade) and one client which still has the enableable wrapper, what happens?

Also what does "write" mean? In this cross version collab case, can I modify an existing value if the enableable type? You can right?

Can I close a subtree that contains it and insert that clone?

Assuming that the ability to insert the type in that location is controlled by the current state of the stored schema, I think we should adopt phrasing more like:

```suggestion
Enablables are allowed types that can be enabled by schema upgrades.
Before being enabled, any attempt to insert or move a node to a location which requires the enablement for its type to be valid will throw an error.
```

Reply:
I like the updated phrasing a lot. I was struggling to make this sound correct :)

>Can I close a subtree that contains it and insert that clone?

I haven't thought about this scenario and I'm not sure if I understand it. What does closing a subtree and cloning it look like in the code?

Reply:
If someone does `mynode.child = TreeBeta.clone(myNode.child);` you might expect it to replace the tree with an identical one, not error if it contains new stuff.

Same for if someone does `mynode.child = TreeAlpha.importVerbose(TreeAlpha.exportVerbose(myNode.child);`

It seems odd if such cases error. It sounds like the intention is for them to error, which is ok for now, but we should make it clear that this approach can cause such errors.